### ISSUE TITLE
DALI light state not updated on scene change

### ISSUE THREAD

**Issue by mrbrdo** [XKNX/knx-integration#3]
I've had a problem pretty consistently where the state of DALI lights is not updated when switching Scenes. When starting off with an off light and then switching through scenes, a light will change on/off or change the brightness, but will just show as "off" in HA, for example.

It seems the scene change only sends a KNX write to the brightness_address, but not the "address" (for state change). It correctly turns on the light, but the state is not updated in HA.

Additionally, I noticed that simply reading from the KNX bus from brightness_address will correctly, automatically update the brightness value in the HA light entity, but the same does not work for state. Reading from the state address (through automation) does not update the state of the light. I think it would be nice if it did.

**Comment by farmio** [comment]
Hi 👋! 
Which DALI gateway are you using? The one I've seen all support proper state objects - if configured they should work fine.

> Reading from the state address (through automation) does not update the state of the light. I think it would be nice if it did.

State should update just as brightness - regardless of a write or response. Maybe you can post your config and a group monitor screenshot so we can find out what is happening there.

**Comment by mrbrdo** [comment]
@farmio I'm using a JUNG DALI 2099 REGHE.
For this device, the status object for brightness and on/off state has to be set as either active signalling object or passive object, as described on page 108 in the [manual](https://downloads.jung.de/public/en/pdf/productdocumentation/en_2099REGHE_td_v2_16062020.pdf). This is kind of annoying as a "passive" object will not signal changes on the bus, while the "active" object will, however the active object cannot be read out. This means the active object won't work correctly on HA boot-up, for example. Because of this I am using "passive" status objects.
I think that because the status object is passive, when HA writes the brightness when I set a Scene, it will not be notified of the change to "on" state, even though that happens internally. HA will still show the light as "off", even though it was turned on with the specified brightness. The states are reflected correctly in ETS group monitor, but aren't updated in HA, specifically when switching Scenes. I tried making an automation which will read out the state after brightness change (write to brightness value address on the bus), which then correctly updates the brightness in HA. I made another one for reading out the on/off state after a turn on/off telegram, but this one does not update the state (even though the read-out value is correct).

**Comment by farmio** [comment]
What flags are set on the status feedback object when you activate "active"?

> The states are reflected correctly in ETS group monitor, but aren't updated in HA, specifically when switching Scenes.

Why can you see the states in ETS group monitor when the status feedback is set to passive - shouldn't it not send a new value then? Or do you only use "passive" for the "on-off-status"?

> I made another one for reading out the on/off state after a turn on/off telegram, but this one does not update the state (even though the read-out value is correct).

It's quite hard to follow that without seeing the group monitor and entity config. Maybe you can add some screenshots.

**Comment by mrbrdo** [comment]
@farmio 
When passive: Communication, Read
When active: Communication, Transmit

When it's passive, I don't see the state update in ETS group monitor, but if I read the value out, it will show me the correct new state. When it's passive, I only see the writes to the switching/dimming address, but there is no signal about state change until I actively read it. I'm using both brightness and on/off as passive objects.

I'm not sure what you'd like to see from the screenshot, can you let me know and I will do it? Basically it's just when it's passive, there is no active signal on the status object, but I can read it. When it's active, it will signal the state change, but if I try to read it, it won't give a result.

**Comment by farmio** [comment]
Have you tried to set it to active and set the read flag?
(Or passive and set the transmit flag - theoretically this should be the same). You should have a readable object that actively transmits new values then.

**Comment by mrbrdo** [comment]
@farmio I didn't know you could do that!
It seems like that worked. I will test it out a bit more this evening and let you know if everything works correctly now.
